### PR TITLE
Fix sync tests on latest WP Develop and VVV

### PR DESCRIPTION
**Pull request by gravityrail**
Fixes many issues with `--testsuite=sync` on VVV with latest wordpress-develop branch.

#### Changes proposed in this Pull Request:

* Set `WP_PLUGIN_DIR` and `ABSPATH` in boostrap.php so that plugin installation and update tests pass. Previously they were being set to the values for the `wordpress-develop` install (which hosts the test classes) but I wanted them to point to the `wordpress-default` install, where Jetpack is being developed.
* Replace reference to require `$test_root . '/includes/listener-loader.php';` with `speed-trap-listener.php`. This file appears to have been renamed recently.
* Don't run `test_add_whitelisted_option_on_init_89` if `init` has already run - it's almost guaranteed to fail. Warn instead so we can fix that test properly in the future.
* Add new `"jetpack_filter_${transient}_value"` filter which lets us inject a fake update into the list of available WP versions, just in case there are no updates against the WP version you're testing on.

#### Testing instructions:
* run `phpunit --testsuite=sync` on a VVV install, with this branch checked out in the `wordpress-default` branch. `wordpress-default` and `wordpress-develop` installs should both be as up-to-date as possible.

#### Proposed changelog entry for your changes:

* Improve test reliability on VVV

### CODE REVIEW

File: tests/php/bootstrap.php [hunk 1]
```diff
@@ -117,4 +121,4 @@ function in_running_uninstall_group() {
 
 // Using the Speed Trap Listener provided by WordPress Core testing suite to expose
 // slowest running tests. See the configuration in phpunit.xml.dist
-require $test_root . '/includes/listener-loader.php';
+require $test_root . '/includes/speed-trap-listener.php';
```
Comment (kraftbj) [inline tests/php/bootstrap.php]: I believe this needs to be reverted back. Different versions of PHPUnit requires different speed-trap-listeners. The loader helps ensure the right one is loaded.

https://github.com/WordPress/wordpress-develop/blob/master/tests/phpunit/includes/listener-loader.php

PHPUnit 7 requires PHP 7.1, so one is loaded if testing on newer PHP; the other for PHP 7 or below.

Comment (gravityrail) [reply tests/php/bootstrap.php]: Yep, my mistake, needed to `svn up` my source tree

**Comment by kraftbj** [comment]
VVV 3.0 is now the latest version so I've tested against that. By default, it includes two standard WP installs. Both master and this branch errors out with:

```
To run Jetpack woocommerce tests, prefix phpunit with JETPACK_TEST_WOOCOMMERCE=1
PHP Notice:  Undefined variable: test_root in /srv/www/wordpress-one/public_html/wp-content/plugins/jetpack/tests/php/bootstrap.php on line 67
PHP Warning:  require(/includes/functions.php): failed to open stream: No such file or directory in /srv/www/wordpress-one/public_html/wp-content/plugins/jetpack/tests/php/bootstrap.php on line 67
PHP Fatal error:  require(): Failed opening required '/includes/functions.php' (include_path='.:/usr/share/php') in /srv/www/wordpress-one/public_html/wp-content/plugins/jetpack/tests/php/bootstrap.php on line 67
```
which makes sense since the tests don't exist.

If I set the `wordpress-trunk` to provision in my `vvv-custom.yml` file, I can get the tests to run fine if I add the right file path in our list for `$test_root`. But, on this branch, that throws a `PHP Notice:  Constant ABSPATH already defined in /srv/www/wordpress-trunk/public_html/wp-tests-config.php on line 7` notice.

File: tests/php/bootstrap.php [hunk 2]
```diff
@@ -16,6 +16,10 @@
  */
 define( 'TESTING_IN_JETPACK', true );
 
+// ensure we load plugins from the correct place
+define( 'WP_PLUGIN_DIR', dirname( dirname( dirname( dirname( __FILE__ ) ) ) ) );
```
Comment (kraftbj) [inline tests/php/bootstrap.php]: This is not needed from my tests with latest VVV. Throws a `PHP Notice:  Constant ABSPATH already defined in /srv/www/wordpress-trunk/public_html/wp-tests-config.php on line 7` error.

**Review by kraftbj — CHANGES_REQUESTED** [review]

**Comment by kraftbj** [comment]
Digging in more, it looks like when VVV moved to 3.0, they missed setting up the `wp-tests-config.php` file. I'm going to leave this as blocked until they work on that so we can test realistically without having to work around that limitation.

**Comment by gravityrail** [comment]
I'm going to close this and upgrade to VVV 3.0 and see if the issues continue

**Comment by kraftbj** [comment]
I'll open a new PR for https://github.com/Automattic/jetpack/pull/12410/commits/ce652dac793773e15c73e04d421e925f595ee301 since we'll need that with 3.0 either way with what you were looking to fix.

**Pull request closed without merging**